plugins/inputs/huawei_routers_telemetry/huawei_routers_telemetry.go

```diff
@@ -0,0 +1,230 @@
+	SetReadBuffer(bytes int) error
+}
+
+type streamSocketListener struct {
+	net.Listener
+	*HuaweiRoutersTelemetry
+
+	sockType string
+
+	connections    map[string]net.Conn
+	connectionsMtx sync.Mutex
+}
+
+type packetSocketListener struct {
+	net.PacketConn
+	*HuaweiRoutersTelemetry
+}
+
+/*
+  Telemetry Decoder.
+
+*/
+func HuaweiTelemetryDecoder(body []byte) *metric.SeriesGrouper {
+	msg := &telemetry.Telemetry{}
+	err := proto.Unmarshal(body[12:], msg)
+	if err != nil {
+		fmt.Println("Error", err)
+		panic(err)
+	}
+	grouper := metric.NewSeriesGrouper()
+	for _, gpbkv := range msg.GetDataGpb().GetRow() {
+		dataTime := gpbkv.Timestamp
+		if dataTime == 0 {
+			dataTime = msg.MsgTimestamp
+		}
+		timestamp := time.Unix(int64(dataTime/1000), int64(dataTime%1000)*1000000)
+		sensorMsg := huawei_sensorPath.GetMessageType(msg.GetSensorPath())
+		err = proto.Unmarshal(gpbkv.Content, sensorMsg)
+		if err != nil {
+			fmt.Println("Sensor Error", err)
+		} else {
+			fields, vals := huawei_sensorPath.SearchKey(gpbkv, msg.GetSensorPath())
+			tags := make(map[string]string, len(fields)+3)
+			tags["source"] = msg.GetNodeIdStr()
+			tags["subscription"] = msg.GetSubscriptionIdStr()
+			tags["path"] = msg.GetSensorPath()
+			// Search for Tags
+			for i := 0; i < len(fields); i++ {
+				tags = huawei_sensorPath.AppendTags(fields[i], vals[i], tags, msg.GetSensorPath())
+			}
+			// Create Metrics
+			for i := 0; i < len(fields); i++ {
+				huawei_sensorPath.CreateMetrics(grouper, tags, timestamp, msg.GetSensorPath(), fields[i], vals[i])
+			}
+		}
+	}
+	return grouper
+}
+
+/*
+  Listen UDP packets and call the telemetryDecoder.
+*/
+func (psl *packetSocketListener) listen() {
+	buf := make([]byte, 64*1024) // 64kb - maximum size of IP packet
+	for {
+		n, _, err := psl.ReadFrom(buf)
+		if err != nil {
+			if !strings.HasSuffix(err.Error(), ": use of closed network connection") {
+				psl.Log.Error(err.Error())
+			}
+			break
+		}
+
+		body, err := psl.decoder.Decode(buf[:n])
+		if err != nil {
+			psl.Log.Errorf("Unable to decode incoming packet: %s", err.Error())
+		}
+		// Inicia el manejo de la telemetria
+		grouper := HuaweiTelemetryDecoder(body)
+		for _, metric := range grouper.Metrics() {
+			psl.AddMetric(metric)
+		}
+
+		if err != nil {
+			psl.Log.Errorf("Unable to parse incoming packet: %s", err.Error())
+
+			continue
+		}
+	}
+}
+
+type HuaweiRoutersTelemetry struct {
+	ServicePort     string        `toml:"service_port"`
+	ReadBufferSize  internal.Size `toml:"read_buffer_size"`
+	ContentEncoding string        `toml:"content_encoding"`
+	wg              sync.WaitGroup
+
+	Log telegraf.Logger
+
+	telegraf.Accumulator
+	io.Closer
+	decoder internal.ContentDecoder
+}
+
+func (sl *HuaweiRoutersTelemetry) Description() string {
+	return "Huawei Telemetry UDP model input Plugin"
+}
+
+func (sl *HuaweiRoutersTelemetry) SampleConfig() string {
+	return `
+  ## UDP Service Port to capture Telemetry
+  service_port = "8080"
```

Comment:
This is probably a sane default, isn't it? So comment this line
```suggestion
  # service_port = "8080"
```

Reply:
Comment added